internals/search_queries.py

```diff
@@ -62,6 +62,14 @@ def single_field_query_async(
     return []
 
   # TODO(jrobbins): Handle ":" operator as substrings for text fields.
+
+  try:
+    # If the field can be set to val, then it can be compared to val.
+    field._validate(val)
```

Comment:
I believe the validation uses **isinstance** to validate. It would not work if val is a list of integer, e.g. '5,6,7,8' where field is **Integer** and val is **String**. It will run **query.filter(field.IN(val_list))**

Reply:
True, but I don't think that was previously supported either.  We only supported the quick-OR syntax with string fields.
I have added a TODO to add support for quick-OR on integer fields.

Reply:
Yea I realized that it was the case. Thanks for adding a comment there!